### DISCUSSION TITLE
PLE Metrics?

### DISCUSSION THREAD

**Discussion by TheAccidentalDBA** [trimble-oss/dba-dash#178] [Ideas]
**EDIT**
I found where PLE's are displayed.  I just don't know how to delete this thread!

I know you can create custom checks, but I was curious if this is something that can be added in future releases?

**Comment by DavidWiseman** [comment]
This is already captured.  

- Select the Instance in the tree
- Open the "Metrics" tab
- You should see a counter for 'Page life Expectancy' in the list.  
- Click the hyperlink in the Counter column to see the graph

Tip: You can enter "page life" in the search box.  Click enter and the metrics will be filtered making it easier to find.

If you want to add any additional counters that are located in the sys.dm_os_performance_counters DMV it's really easy to do so.  If you look at the PerformanceCounters.xml file in the application folder you will see how it's setup.  To add new counters - instead of changing this file, make a copy called "PerformanceCountersCustom.xml" and edit that file instead.  (Otherwise your changes will be overwritten when you upgrade to a new version)

You could also add any custom application metrics by creating a stored proc to return whatever metrics you want.  The documentation is [here](https://github.com/trimble-oss/dba-dash/blob/main/Docs/OSPerformanceCounters.md).

Note: Custom metrics are different from custom checks.  Metrics will be collected every 1min by default along with the performance counters - they show up in the Metrics tab.  Custom checks appear on the Summary page and they are a way to perform additional validation checks (Maybe a check for failed SSRS subscriptions or something).  They run every 1hr by default.

Hope this helps!

**Comment by DavidWiseman** [comment]
If you are troubleshooting memory issues you could also look at the "Memory" tab.  This will give you a breakdown of memory usage by memory clerk.  There is a "Memory Performance Counters" sub tab that shows a filtered list of the performance counters available in the Metrics tab.
